miner/miner.go

```diff
@@ -2,29 +2,98 @@ package miner
 
 import (
 	"math"
+	"sync"
 	"time"
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/ubclaunchpad/cumulus/blockchain"
 	"github.com/ubclaunchpad/cumulus/consensus"
 )
 
+// currentlyMining is a flag to control the miner.
+var currentlyMining bool
+
+// currentlyMiningLock is a read/write lock to change the Mining flag.
+var currentlyMiningLock sync.RWMutex
+
+const (
+	// MiningSuccessful is returned when the miner mines a block.
+	MiningSuccessful = iota
```

Comment:
You only need to set the first constant to `iota`, the compiler will fill in the following ones on its own (see https://golang.org/ref/spec#Constant_declarations and the section below).